packages/Uploader/src/Uploader.js

```diff
@@ -0,0 +1,219 @@
+import React from "react";
+import PropTypes from "prop-types";
+import uuidv4 from "uuid/v4";
+import stylers from "@paprika/stylers/";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import { inputFileStyles, labelStyles, containerStyles } from "./Uploader.styles";
+import { getFiles } from "./helpers";
+import ProgressBar from "./components/ProgressBar";
+import types from "./types";
+import useDragAndDropEvents from "./useDragAndDropEvents";
+import useProcessFiles from "./useProcessFiles";
+
+const oneMebibyte = 1048576;
+
+const propTypes = {
+  /**
+    Accessible message for the input[type="file"].
+  */
+  a11yText: PropTypes.string,
+  /**
+    An array of string describing the allowed file types for the uploader ex. ["image/*", ".pdf", ".doc", "docx"]
+    reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#Unique_file_type_specifiers.
+  */
+  acceptableFileTypes: PropTypes.arrayOf(PropTypes.string),
+  /**
+    When false the uploader only accept one file per upload.
+   */
+  allowMultipleFile: PropTypes.bool,
+  /**
+    children function received the following parameter which you can use to build you own UI:
+
+    FileInput,   // the input[type="file"] element ready to be consumed
+    files,       // a enhance list of the user selected files
+    hasFinished, // is true when all files have been processed
+    isDisabled,  // describe the status of the component
+    isDragLeave, // will be true if a dragOver event occurred and leave the droppable area
+    isDragOver,  // True when detecting the user is draggin files over the droppable area
+    removeFile,  // removeFile(key) will remove a file from the files list, this function doesn't work while the file is on PROCESSING state.
+    cancelFile,  // cancelFile(key) stop the request cycle keep in mind that if it's on WAITINGFORSERVER state the server might save the file even if the request has been cancel
+    upload,      // upload() give you the option to manually upload the files if you are decide to not use hasAutoupload
```

Comment:
This is a lot for inline prop comments.  Can we just leave it to the README, or even reference the README here for more info?